### PR TITLE
Perception: avoid unnecessary copy.

### PR DESCRIPTION
**Pull request by csukuangfj**

### CODE REVIEW

File: modules/perception/lib/base/registerer.h [hunk 1]
```diff
@@ -138,7 +138,7 @@ bool GetRegisteredClasses(
       FactoryMap &map = perception::GlobalFactoryMap()[#base_class];  \
       FactoryMap::iterator iter = map.find(name);                     \
       if (iter == map.end()) {                                        \
-        for (auto c : map) {                                          \
+        for (const auto &c : map) {                                   \
```
Comment (startcode) [inline modules/perception/lib/base/registerer.h]: Nice catch. Thanks for the change.
 But can you also replace `map` by a better variable name
Thanks.

Comment (csukuangfj) [reply modules/perception/lib/base/registerer.h]: @startcode
thanks for the comment.

I don't think map is a bad name in this context. If you can suggest a new name, I would like to replace it.

Comment (csukuangfj) [reply modules/perception/lib/base/registerer.h]: `object_factory_map` is too long and it exceeds 80 columns per row. In addition, the backslash
at the end of the line will not be aligned.

Comment (startcode) [reply modules/perception/lib/base/registerer.h]: `map`  fine. it a minor thing anyway.

**Pull request merged**